### PR TITLE
chore(service-provider-server) remove returnNonCachedInstance option

### PR DESCRIPTION
**Pull request by aherlihy**
See https://jira.mongodb.org/browse/NODE-2931

### CODE REVIEW

File: packages/service-provider-server/src/cli-service-provider.integration.spec.ts
Comment (addaleax) [inline packages/service-provider-server/src/cli-service-provider.integration.spec.ts:682]: I can see that they were failing in CI, but that surprises me, because these are not supposed to be affected by the driver’s caching at all and instead are supposed to test our *own* caching, and that the tests failed because `db()` returned the same `Db` instance for different options does seem a bit concerning to me … do you know more about this?

Comment (addaleax) [reply packages/service-provider-server/src/cli-service-provider.integration.spec.ts:682]: It looks like for now the caching feature is still alive and well in the driver’s master branch: https://github.com/mongodb/node-mongodb-native/blob/cb9ee9e6175a6654c3c300801884e4a3c3a653ac/src/mongo_client.ts#L404

File: packages/service-provider-server/src/cli-service-provider.ts
Comment (addaleax) [inline packages/service-provider-server/src/cli-service-provider.ts:280]: I think for now what we could probably do is to remove the `as any` along with `: DbOptions`? That should let typescript figure out that this type is “`DbOptions` + this one other thing that is not mentioned there but that the `.db()` method still accepts”

File: packages/service-provider-server/src/cli-service-provider.integration.spec.ts
Comment (aherlihy) [reply packages/service-provider-server/src/cli-service-provider.integration.spec.ts:682]: Ah, it seems this is in code review https://jira.mongodb.org/browse/NODE-2931 so not quite in master yet, so that was a little presumtious of me to make this PR

File: packages/service-provider-server/src/cli-service-provider.ts
Comment (aherlihy) [reply packages/service-provider-server/src/cli-service-provider.ts:280]: I think we can just put this PR on hold until it's merged into node master, I didn't realize it hadn't been already done. Then we can revert some of the tests so that it makes sure the behavior's still happening

Comment (aherlihy) [reply packages/service-provider-server/src/cli-service-provider.ts:280]: Until this is merged: https://github.com/mongodb/node-mongodb-native/pull/2672

**Review by addaleax — APPROVED** [review]

**Pull request merged**
